spec/stopNodes_spec.js

```diff
@@ -0,0 +1,138 @@
+"use strict";
+
+const parser = require("../src/parser");
+const validator = require("../src/validator");
+const he = require("he");
+
+describe("XMLParser", function() {
+    it("1a. should support single stopNode", function() {
+        const xmlData = `<issue><title>test 1</title><fix1><p>p 1</p><div class="show">div 1</div></fix1></issue>`;
+        const expected = {
+            "issue": {
+                "title": "test 1",
+                "fix1": "<p>p 1</p><div class=\"show\">div 1</div>"
+            }
+        };
+
+        let result = parser.parse(xmlData, {
+            attributeNamePrefix: "",
+            ignoreAttributes:    false,
+            parseAttributeValue: true,
+            stopNodes: ["fix1"]
+        });
+
+        //console.log(JSON.stringify(result,null,4));
+        expect(result).toEqual(expected);
+
+        result = validator.validate(xmlData);
+        expect(result).toBe(true);
+    });
+
+    it("1b. 3. should support more than one stopNodes, with or without attr", function() {
+        const xmlData = `<issue><title>test 1</title><fix1 lang="en"><p>p 1</p><div class="show">div 1</div></fix1><fix2><p>p 2</p><div class="show">div 2</div></fix2></issue>`;
```

Comment:
Please write a test when fix2 is inside fix1. In this case, fix2 should be empty in the result.

Reply:
Done. Thanks for the suggestion.